packages/flutter/lib/src/widgets/basic.dart

```diff
@@ -2164,6 +2165,7 @@ class AsyncImage extends StatelessWidget {
   @override
   Widget build(BuildContext context) {
     return new RawImageResource(
+      key: new ObjectKey(provider),
```

Comment:
This must be a ValueKey<ImageProvider>, not an ObjectKey, otherwise even ImageProviders that are == will cause the underlying object to be replaced.